MonMaraicher/MapView.swift

```diff
@@ -10,9 +10,32 @@ import MapKit
 
 struct MapView: View {
     
+    @StateObject private var viewModel = MapViewModel()
+    
     var body: some View {
-        
-        Text("Hello world!")
+        Map(position: $viewModel.userPosition, selection: $viewModel.selectedFarmerPlace) {
+            
+            ForEach(viewModel.allFarmerPlaces, id: \.id) { place in
+                Marker(place.name, coordinate: CLLocationCoordinate2D(latitude: place.location.latitude, longitude: place.location.longitude))
+                    .tag(place)
+                    .tint(.green)
+            }
+            
+            UserAnnotation()
+            
+        }
+        .onChange(of: viewModel.selectedFarmerPlace, { oldValue, newValue in
+            viewModel.showDetails = newValue != nil
+        })
+        .sheet(isPresented: $viewModel.showDetails, content: {
```

Comment:
Why not using the other .sheet modifier: https://developer.apple.com/documentation/musickit/artworkimage/sheet(item:ondismiss:content:)

I let you try it and see the benefits ;) Tell me if you need some help :)